bokeh/io.py

```diff
@@ -439,17 +434,13 @@ def _get_save_args(state, filename, resources, title):
 
     return filename, resources, title
 
-def _save_helper(obj, filename, resources, title, validate):
+def _save_helper(obj, state, filename, resources, title, validate):
     remove_after = False
-    if isinstance(obj, LayoutDOM):
-        if obj.document is None:
-            Document().add_root(obj)
```

Comment:
This is cause of the noted bug (themes not being applied). The solution is the get the document from io._state and not create a new one.